src/components/sbb-header-action/sbb-header-action.scss

```
@@ -0,0 +1,97 @@
+// Default component properties, defined for :host. Properties which can not
+// travel the shadow boundary are defined through this mixin
+@include host-component-properties;
+
+// Displays the action item's text and corrects spacings.
+@mixin expandFromBreakpoint($fromValue) {
+  @include mq($from: $fromValue) {
+    --sbb-header-action-text-display: block;
+    --sbb-header-action-icon-margin: var(--sbb-spacing-fixed-2x);
+    --sbb-header-action-padding-inline: var(--sbb-spacing-fixed-5x);
+    --sbb-header-action-min-width: inherit;
+  }
+}
+
+:host {
+  --sbb-header-action-text-display: none;
+  --sbb-header-action-icon-margin: 0;
+  --sbb-header-action-padding-inline: 0;
+  --sbb-header-action-background-color: var(--sbb-color-white-default);
+  --sbb-header-action-min-height: var(--sbb-size-button-m-min-height);
+  --sbb-header-action-min-width: var(--sbb-header-action-min-height);
+
+  display: flex;
+}
+
+:host([expand-from='zero']) {
+  @include expandFromBreakpoint(zero);
+}
+
+:host([expand-from='micro']) {
+  @include expandFromBreakpoint(micro);
+}
+
+:host([expand-from='small']) {
+  @include expandFromBreakpoint(small);
+}
+
+:host([expand-from='medium']) {
+  @include expandFromBreakpoint(medium);
+}
+
+:host([expand-from='large']) {
+  @include expandFromBreakpoint(large);
+}
+
+:host([expand-from='wide']) {
+  @include expandFromBreakpoint(wide);
+}
+
+:host([expand-from='ultra']) {
+  @include expandFromBreakpoint(ultra);
+}
```

Here the SAAS `@each` at-rule could be used:
```suggestion
$breakpoints: 'zero', 'micro', 'small', 'medium', 'large', 'wide', 'ultra';

@each $breakpoint in $breakpoints {
  :host([expand-from='#{$breakpoint}']) {
    @include expandFromBreakpoint($breakpoint);
  }
}
```